### PR TITLE
Reliability fixes for benchmarking work and 0.8.2 releasenotes

### LINKED ISSUE #785: RDKitToolkitWrapper double bond stereochemistry checker too strict
**Describe the bug**
`C=C` and `C=N` bonds with equivalent substituents on one atom are being flagged as having missing stereochemical information when loading from 3D SDF using RDKitToolkitWrapper.  

**To Reproduce**

```
from openforcefield.topology import Molecule
# This molecule has no stereocenters
mol = Molecule.from_smiles('C=C')
mol.generate_conformers()
mol.to_file('xxx.sdf', file_format='sdf')
mol2 = Molecule.from_file('xxx.sdf')
```

> ---------------------------------------------------------------------------
> UndefinedStereochemistryError             Traceback (most recent call last)
> <ipython-input-1-aa21b0d0c6fb> in <module>()
>       3 mol.generate_conformers()
>       4 mol.to_file('xxx.sdf', file_format='sdf')
> ----> 5 mol2 = Molecule.from_file('xxx.sdf')
> 
> /Users/jeffreywagner/projects/OpenForceField/openforcefield/openforcefield/topology/molecule.py in from_file(cls, file_path, file_format, toolkit_registry, allow_undefined_stereo)
>    4070                 file_format=file_format,
>    4071                 allow_undefined_stereo=allow_undefined_stereo,
> -> 4072                 _cls=cls,
>    4073             )
>    4074         elif hasattr(file_path, "read"):
> 
> /Users/jeffreywagner/projects/OpenForceField/openforcefield/openforcefield/utils/toolkits.py in from_file(self, file_path, file_format, allow_undefined_stereo, _cls)
>    2830                 Chem.SetAromaticity(rdmol, Chem.AromaticityModel.AROMATICITY_MDL)
>    2831                 mol = self.from_rdkit(
> -> 2832                     rdmol, allow_undefined_stereo=allow_undefined_stereo, _cls=_cls
>    2833                 )
>    2834                 mols.append(mol)
> 
> /Users/jeffreywagner/projects/OpenForceField/openforcefield/openforcefield/utils/toolkits.py in from_rdkit(self, rdmol, allow_undefined_stereo, _cls)
>    3439             rdmol,
>    3440             raise_warning=allow_undefined_stereo,
> -> 3441             err_msg_prefix="Unable to make OFFMol from RDMol: ",
>    3442         )
>    3443 
> 
> /Users/jeffreywagner/projects/OpenForceField/openforcefield/openforcefield/utils/toolkits.py in _detect_undefined_stereo(cls, rdmol, err_msg_prefix, raise_warning)
>    4182             else:
>    4183                 msg = "Unable to make OFFMol from RDMol: " + msg
> -> 4184                 raise UndefinedStereochemistryError(msg)
>    4185 
>    4186     @staticmethod
> 
> UndefinedStereochemistryError: Unable to make OFFMol from RDMol: Unable to make OFFMol from RDMol: RDMol has unspecified stereochemistry. RDMol name: Bonds with undefined stereochemistry are:
>  - Bond 0 (atoms 0-1 of element (C-C)

**Proposed solution**

The code below is my best attempt at detecting which bonds are stereogenic, starting from an SDF. We can use other code to ensure that all stereogenic bonds have defined stereochemistry.

(I'm posting a lengthy writeup since this may be of interest to others)

Using molecule `C=CC/C=C(/Cl)`. Note that the double bond on the right IS stereogenic, and the one on the left is NOT stereogenic

image

```
# Here we have RDKit make a molecule with two double bonds, one of which is stereogenic, and write it to SDF

from rdkit import Chem
from rdkit.Chem import AllChem
rdmol = Chem.MolFromSmiles('C=CC/C=C(/Cl)')
#rdmol = Chem.MolFromSmiles('C=C')
#rdmol = Chem.MolFromSmiles('FC=CCl')

Chem.SanitizeMol(rdmol,
                 Chem.SANITIZE_ALL ^ Chem.SANITIZE_ADJUSTHS ^ Chem.SANITIZE_SETAROMATICITY,
                )
Chem.AssignStereochemistry(rdmol)

Chem.FindPotentialStereoBonds(rdmol)

rdmol = Chem.AddHs(rdmol)

AllChem.EmbedMultipleConfs(
    rdmol,
    numConfs=1,
    pruneRmsThresh=1,
    randomSeed=1,
)
for bond in rdmol.GetBonds():
    print('before file write', bond.GetStereo())
    
writer = Chem.SDWriter('xxx_rd.sdf')
writer.write(rdmol)
writer.close()
```

> before file write STEREONONE <-- Here's the non-stereogenic bond
> before file write STEREONONE
> before file write STEREONONE
> before file write STEREOE <-- Here's the actual stereogenic bond
> before file write STEREONONE
> before file write STEREONONE
> before file write STEREONONE
> before file write STEREONONE
> before file write STEREONONE
> before file write STEREONONE
> before file write STEREONONE
> before file write STEREONONE

```
# Next we load it from SDF, using the same code as OFFTK currently does
rdmol2 = None
for mol in Chem.ForwardSDMolSupplier(open('xxx_rd.sdf','rb'), sanitize=False):
    rdmol2 = mol
    
# Here is the similar sanitization code to what OFFTK runs
Chem.AssignAtomChiralTagsFromStructure(rdmol2)

Chem.SanitizeMol(
    rdmol2,
    (
        Chem.SANITIZE_ALL
        ^ Chem.SANITIZE_SETAROMATICITY
        ^ Chem.SANITIZE_ADJUSTHS
        ^ Chem.SANITIZE_CLEANUPCHIRALITY
        ^ Chem.SANITIZE_KEKULIZE
    ),
)
    
Chem.FindPotentialStereoBonds(rdmol2)
Chem.AssignCIPLabels(rdmol2)


for bond in rdmol2.GetBonds():
    print('After finding potential stereo bonds', bond.GetStereo())
```

> After finding potential stereo bonds STEREOANY <-- Here's the non-stereogenic bond
> After finding potential stereo bonds STEREONONE
> After finding potential stereo bonds STEREONONE
> After finding potential stereo bonds STEREOANY <-- Here's the actual stereogenic bond
> After finding potential stereo bonds STEREONONE
> After finding potential stereo bonds STEREONONE
> After finding potential stereo bonds STEREONONE
> After finding potential stereo bonds STEREONONE
> After finding potential stereo bonds STEREONONE
> After finding potential stereo bonds STEREONONE
> After finding potential stereo bonds STEREONONE
> After finding potential stereo bonds STEREONONE

```
# Trying again with cleanIt=True
Chem.FindPotentialStereoBonds(rdmol2, cleanIt=True)
Chem.AssignCIPLabels(rdmol2)

for bond in rdmol2.GetBonds():
    print('After finding potential stereo bonds with cleanIt=True', bond.GetStereo())
```

> After finding potential stereo bonds with cleanIt=True STEREOANY <-- Here's the non-stereogenic bond
> After finding potential stereo bonds with cleanIt=True STEREONONE
> After finding potential stereo bonds with cleanIt=True STEREONONE
> After finding potential stereo bonds with cleanIt=True STEREOANY <-- Here's the actual stereogenic bond
> After finding potential stereo bonds with cleanIt=True STEREONONE
> After finding potential stereo bonds with cleanIt=True STEREONONE
> After finding potential stereo bonds with cleanIt=True STEREONONE
> After finding potential stereo bonds with cleanIt=True STEREONONE
> After finding potential stereo bonds with cleanIt=True STEREONONE
> After finding potential stereo bonds with cleanIt=True STEREONONE
> After finding potential stereo bonds with cleanIt=True STEREONONE
> After finding potential stereo bonds with cleanIt=True STEREONONE

```
# Manually wiping existing stereo and trying again with cleanIt=True
for bond in rdmol2.GetBonds():
    bond.SetStereo(Chem.BondStereo.STEREONONE)

Chem.FindPotentialStereoBonds(rdmol2, cleanIt=True)
Chem.AssignCIPLabels(rdmol2)

    
for bond in rdmol2.GetBonds():
    print('After unset chirality and re-percieve', bond.GetStereo())
```

> After unset chirality and re-percieve STEREONONE <-- Here's the non-stereogenic bond
> After unset chirality and re-percieve STEREONONE
> After unset chirality and re-percieve STEREONONE
> After unset chirality and re-percieve STEREOANY <-- Here's the actual stereogenic bond
> After unset chirality and re-percieve STEREONONE
> After unset chirality and re-percieve STEREONONE
> After unset chirality and re-percieve STEREONONE
> After unset chirality and re-percieve STEREONONE
> After unset chirality and re-percieve STEREONONE
> After unset chirality and re-percieve STEREONONE
> After unset chirality and re-percieve STEREONONE
> After unset chirality and re-percieve STEREONONE

**System details**

Mac OSX 10.14
```
rdkit                     2020.09.1        py37h3dce739_0    conda-forge
openforcefield 0.8.1
```
(behavior reproduces with several earlier versions of both toolkits)

### PR DESCRIPTION
**Pull request by j-wags**
- [x] Fixes #785 
- [x] Fix Molecule copy initializer (`offmol2 = Molecule(offmol)`) so that `offmol2.properties` doesn't point to the same dict as `offmol.properties`
- [x] Add [tests](https://github.com/openforcefield/openforcefield/tree/master/openforcefield/tests)
- [x] Update docstrings/[documentation](https://github.com/openforcefield/openforcefield/tree/master/docs), if applicable
- [x] [Lint](https://open-forcefield-toolkit.readthedocs.io/en/latest/developing.html#style-guide) codebase
- [x] Update [changelog](https://github.com/openforcefield/openforcefield/blob/master/docs/releasehistory.rst)

### CODE REVIEW

**Comment by j-wags** [comment]
I've put on my thinking hat, and I think the test I've added for #785 is appropriate, and that other existing tests ensure the continued detection of ACTUAL undefined stereo bonds. 

I've also modified an existing test to ensure that the molecule copy initializer DOES produce a unique properties dict in memory. 

Typing up releasenotes now. This should be ready for review.

File: openforcefield/tests/test_toolkits.py [hunk 1]
```diff
@@ -2025,6 +2025,16 @@ def test_write_sdf_no_charges(self):
         # out "n/a" (or another placeholder) in the partial charge block atoms without charges.
         assert ">  <atom.dprop.PartialCharge>" not in sdf_text
 
+    def test_read_ethene_sdf(self):
+        """
+        Test that RDKitToolkitWRapper can load an ethene molecule without complaining about bond stereo
+        See https://github.com/openforcefield/openforcefield/issues/785
```
Comment (mattwthompson) [inline openforcefield/tests/test_toolkits.py:2031]:
```suggestion
        Test that RDKitToolkitWrapper can load an ethene molecule without complaining about bond stereo.
        See https://github.com/openforcefield/openforcefield/issues/785
```

**Review by mattwthompson — APPROVED** [review]
LGTM per our discussion

Non-blockers that we noted
* some of our molecule setters do no validation
* there is no `Bond.stereochemistry` setter
* bonds in rdmols with `STEREOANY` are not processed and fall back to the default None

**Comment by j-wags** [comment]
>    some of our molecule setters do no validation

Having had to do some actual work with the toolkit recently, the lack of validation doesn't seem to be the worst thing (the really annoying thing is excessively strict validation or needing to access a private method to get stuff done)

>    there is no Bond.stereochemistry setter

I'm wrapping my head around this. I... think we should allow this, but the OFF Toolkit's current view of molecules is such that a bond's stereochemistry is structurally just as important as its bond order. Then I looked, and we actually DO expose a bond order setter. So, it may not be intentional, but it would at least be good if our API is consistent about this sort of thing. 

>   bonds in rdmols with STEREOANY are not processed and fall back to the default None

This was something I had to puzzle out for a while in the course of fixing this issue. I've added a comment in `from_rdkit` about this assumption so future folks won't be so confused.

**Pull request merged**